### PR TITLE
[Gardening] Add information on building and running unit tests

### PR DESCRIPTION
**Pull request by CodaFi**
@jrose-apple I think you were the last one to touch this, so could you give it a once-over?

### CODE REVIEW

File: docs/Testing.rst [hunk 1]
```diff
@@ -113,6 +117,16 @@ For every target above, there are variants for different optimizations:
   ``check-swift-non-executable-iphoneos-arm64``) -- runs tests not marked with
   ``executable_test`` in ``-Onone`` mode.
 
+To run the unit tests, they must first be built with the following command
+executed in your build directory:
+
+.. code-block:: bash
+
+    % ninja SwiftUnitTests
```
Comment (modocache) [inline docs/Testing.rst]: The rest of this document refers to targets without also referencing specific build tools. `SwiftUnitTests` could also be built with `make SwiftUnitTests`, after all.

My suggestion would be to mirror the build invocation for `check-swift-macosx-x86_64` above, and recommend `cmake --build build/Ninja-ReleaseAssert/swift-macosx-x86_64 -- SwiftUnitTests`.

**Review by modocache — APPROVED** [review]
Awesome, thanks! This was one of the first pieces of documentation I read when starting to contribute, thank you for improving it. I had one tiny suggestion.

File: docs/Testing.rst (hunk 1 above)
Comment (CodaFi) [inline docs/Testing.rst]: Oh, fantastic.  I was looking for this.  Thank you!

File: docs/Testing.rst [hunk 2]
```diff
@@ -91,6 +95,10 @@ Besides ``check-swift``, other targets are also available. Here's the full list:
 
   Runs all tests (primary, validation, and long).
 
+* ``SwiftUnitTests``
+
+  Runs all unit tests.
```
Comment (jrose-apple) [inline docs/Testing.rst]: I'm pretty sure this only *builds* the unit tests. They still have to be invoked, either individually or through lit.

Comment (CodaFi) [inline docs/Testing.rst]: Done.

**Comment by CodaFi** [comment]
@swift-ci please smoke test and merge.

File: docs/Testing.rst [hunk 3]
```diff
@@ -91,6 +95,11 @@ Besides ``check-swift``, other targets are also available. Here's the full list:
 
   Runs all tests (primary, validation, and long).
 
+* ``SwiftUnitTests``
+
+  Builds all unit tests.  Executables are located under
+  ``${SWIFT_BUILD_ROOT}/unittests`` and must be run individually.
```
Comment (jrose-apple) [inline docs/Testing.rst]: I think it's worth mentioning that they *can* be run individually but are also run by `check-swift` by default (which they are).

**Pull request merged**